src/pages/Donate/Donate.js

```diff
@@ -1,17 +1,23 @@
-import React from 'react';
+import React, { useRef } from 'react';
 import { useDispatch } from 'react-redux';
-import { openModal } from './../../actions/index';
+import { openModal } from 'actions/index';
 
-import CustomThankYou from './../../components/Custom/CustomThankYou/CustomThankYou';
+import Button from 'components/Button/Button';
+import CustomThankYou from 'components/Custom/CustomThankYou/CustomThankYou';
 
 const Donate = () => {
   const dispatch = useDispatch();
   const thankYou = <CustomThankYou />
+  const buttonRef = useRef(null);
 
   return (
     <div>
       <h4>Donate</h4>
-      <button onClick={() => dispatch(openModal(thankYou))}>Donate here!</button>
+      <Button type='filled' buttonRef={buttonRef} onClick={() => dispatch(openModal(thankYou))} icon>Opens Modal</Button>
+      <Button type='outlined' buttonRef={buttonRef} onClick={() => console.log('hello')} icon>Say Hello</Button>
+      <Button type='text' buttonRef={buttonRef} onClick={() => console.log('hello')} icon>Button Text</Button>
+      <Button link='https://huoa.org/contact' buttonRef={buttonRef}>Homepage</Button>
+      <Button link='https://www.google.com/' buttonRef={buttonRef}>Google</Button>
```

Comment:
lets just pass `ref={someRef}` or does react not like this for some reason?

Reply:
I get an error for passing ref into a functional component. See screenshot: 

<img width="853" alt="Screen Shot 2020-08-24 at 12 36 34 PM" src="https://user-images.githubusercontent.com/18355766/91088212-75df9280-e606-11ea-9c2f-05ce25894a78.png">

I believe either useRef or forwardRef would solve the error. 

https://github.com/reactjs/reactjs.org/issues/2120